pkg/pillar/cmd/domainmgr/vtfb.go

```diff
@@ -0,0 +1,322 @@
+
+// Constants taken from Linux kernel header: linux/vt.h
+//
+//revive:disable:var-naming
+const (
+	VT_OPENQRY    = 0x5600 // Query for a free Virtual Terminal
+	VT_ACTIVATE   = 0x5606 // Activate Virtual Terminal
+	VT_WAITACTIVE = 0x5607 // Wait for Virtual Terminal
+)
+
+// Global list of Virtual Terminal (VT) consoles
+var deviceVTs []vtInfo
+
+// Execute ioctl(). In case of error, close fd before return
+func doIOCTL(fd int, req uint, value int) error {
+	err := unix.IoctlSetInt(fd, req, value)
+	if err != nil {
+		if err := unix.Close(fd); err != nil {
+			log.Errorf("Cannot close file descriptor: %v", err)
+		}
+		return err
+	}
+	return nil
+}
+
+// Read from a file
+func sysfsRead(file string) (string, error) {
+	contents, err := os.ReadFile(file)
+	if err != nil {
+		return "", err
+	}
+	return string(contents), nil
+}
+
+// Write to a file
+func sysfsWrite(file string, contents string) error {
+	fp, err := os.OpenFile(file, os.O_RDWR, 0)
+	if err != nil {
+		return err
+	}
+	_, err = fp.WriteString(contents)
+	if err != nil {
+		if err := fp.Close(); err != nil {
+			log.Errorf("Cannot close file descriptor: %v", err)
+		}
+		return err
+	}
+	return fp.Close()
+}
+
+// Unbind video framebuffer from framebuffer driver
+func (fbDrv fbDriver) unbind() error {
+	return sysfsWrite(path.Join(fbDrv.sysfs, "unbind"), fbDrv.name)
+}
+
+// Bind video framebuffer to framebuffer driver
+func (fbDrv fbDriver) bind() error {
+	return sysfsWrite(path.Join(fbDrv.sysfs, "bind"), fbDrv.name)
+}
+
+// Get a list of all framebuffer drivers available at sysfs
+func getFBDrvs() ([]fbDriver, error) {
+	fbdir := "/sys/bus/platform/drivers"
+	sysfb := os.DirFS(fbdir)
+	fbDrvs, err := fs.Glob(sysfb, "*-framebuffer")
+
+	if err != nil {
+		return nil, err
+	}
+
+	var drvs []fbDriver
+	for _, v := range fbDrvs {
+		drv := fbDriver{
+			name:  fmt.Sprintf("%s.0", v),
+			sysfs: path.Join(fbdir, v),
+		}
+		drvs = append(drvs, drv)
+	}
+
+	return drvs, nil
+}
+
+// Unbind all framebuffer drivers
+// Our kernel has FRAMEBUFFER_CONSOLE_DETECT_PRIMARY enabled, which
+// means that framebuffer console will automatically select the
+// primary display device (if supported by the architecture).
+// Otherwise, the framebuffer console will always select the first
+// framebuffer driver that is loaded (default behavior). However,
+// this can be changed by fbcon=map: kernel boot option. In
+// a nutshell, so far we don't have a reliable way to detect which
+// framebuffer driver needs to be unbound, so we just try to unbind all
+// available drivers without care with errors.
+func fbUnbindAll() error {
+	fbDrvs, err := getFBDrvs()
+	if err != nil {
+		return err
+	}
+	for _, fb := range fbDrvs {
+		// If the driver is not bound, the unbind will fail (for now we
+		// don't care, let's just log a warning)
+		log.Noticef("Unbind framebuffer driver: %v\n", fb.name)
+		if err := fb.unbind(); err != nil {
+			log.Warnf("Fail to unbind framebuffer driver %v: %v", fb.name, err)
```

Comment:
```suggestion
			log.Warnf("Fail to unbind framebuffer driver %s: %v", fb.name, err)
```
because somebody in the future might think that it is actually more useful for `fb.name` to be of type `[]byte`